### PR TITLE
Feature/login

### PR DESCRIPTION
**Pull request by Yshah112**

### CODE REVIEW

File: DashClothes/settings.py [hunk 1]
```diff
@@ -27,6 +28,27 @@
 
 ALLOWED_HOSTS = []
 
+SITE_ID = 1
+
+SOCIALACCOUNT_PROVIDERS = {
```
Comment (palakparmartntra) [inline DashClothes/settings.py]: is there any need to add this ? @Yshah112 @rinkishivani

File: DashClothes/settings.py [hunk 2]
```diff
@@ -129,3 +158,25 @@
...
+EMAIL_USE_TLS = True
+EMAIL_HOST_USER = 'yshah112002@gmail.com'
+EMAIL_HOST_PASSWORD = 'qmnf nmvl skvq cwur'
+
+
+ACCOUNT_EMAIL_REQUIRED = True #new
+ACCOUNT_EMAIL_VERIFICATION = 'mandatory'
+ACCOUNT_AUTHENTICATED_LOGIN_REDIRECTS=True
+LOGIN_REDIRECT_URL = 'homepage'
+# LOGOUT_REDIRECT_URL = 'account_login'
```
Comment (palakparmartntra) [inline DashClothes/settings.py]: Remove unnecessary commented code. @Yshah112

File: DashClothes/urls.py [hunk 3]
```diff
@@ -19,5 +19,10 @@
 
 urlpatterns = [
     path('admin/', admin.site.urls),
+    path('accounts/', include('allauth.urls')),
+    path('app/',include('accounts.urls')),
+
 
 ]
+# breakpoint()
```
Comment (palakparmartntra) [inline DashClothes/urls.py]: also here.

File: DashClothes/urls.py [hunk 4]
```diff
@@ -19,5 +19,10 @@
 
 urlpatterns = [
     path('admin/', admin.site.urls),
+    path('accounts/', include('allauth.urls')),
+    path('app/',include('accounts.urls')),
```
Comment (palakparmartntra) [inline DashClothes/urls.py]: you can give an empty path for all auth URLs. 
why give an "app" URL path for custom project app URLs? i think it's a bit confusing. @Yshah112
cc. @lalitvasoya @rinkishivani

File: accounts/forms.py [hunk 5]
```diff
@@ -0,0 +1,7 @@
+from django import forms
+from django.contrib.auth.models import User
+
+class UpdateAccountForm(forms.ModelForm):
```
Comment (palakparmartntra) [inline accounts/forms.py]: where did you use this form ?

File: accounts/models.py [hunk 6]
```diff
@@ -11,10 +11,11 @@ class User(AbstractUser):
     created_at = models.DateTimeField(auto_now_add=True)
 
     def __str__(self):
-        self.username
+        return self.username
 
 
 class Address(models.Model):
+
```
Comment (palakparmartntra) [inline accounts/models.py:20]: Add docstrings in both models here.

File: DashClothes/settings.py [hunk 7]
```diff
@@ -129,3 +158,25 @@
...
 AUTH_USER_MODEL = "accounts.User"
+
+EMAIL_BACKEND = "django.core.mail.backends.filebased.EmailBackend"
+EMAIL_FILE_PATH = BASE_DIR / "sent_emails"
+
+EMAIL_BACKEND = 'django.core.mail.backends.smtp.EmailBackend'
+EMAIL_HOST = 'smtp.gmail.com'
+EMAIL_PORT = 587
+EMAIL_USE_TLS = True
+EMAIL_HOST_USER = 'yshah112002@gmail.com'
```
Comment (lalitvasoya) [inline DashClothes/settings.py:163]: Keep this information in `.env` @Yshah112

File: accounts/views.py [hunk 8]
```diff
@@ -1,3 +1,15 @@
...
+from django.shortcuts import render, redirect
+from allauth.account.forms import LoginForm, ResetPasswordForm
+from allauth.account.views import LogoutView
+from django.views.generic import UpdateView
+from .models import *
+from .forms import *
+
 
 # Create your views here.
+def homepage(request):
```
Comment (lalitvasoya) [inline accounts/views.py]: Use template class base view @Yshah112

File: accounts/models.py [hunk 9]
```diff
@@ -11,10 +11,11 @@ class User(AbstractUser):
     created_at = models.DateTimeField(auto_now_add=True)
 
     def __str__(self):
```
Comment (lalitvasoya) [inline accounts/models.py:15]: Add doc string here @Yshah112  @rinkishivani

File: DashClothes/settings.py (hunk 2 above)
Comment (Yshah112) [reply DashClothes/settings.py]: removed

File: DashClothes/urls.py (hunk 3 above)
Comment (Yshah112) [reply DashClothes/urls.py]: removed

File: accounts/forms.py (hunk 5 above)
Comment (Yshah112) [reply accounts/forms.py]: i had removed that

File: DashClothes/urls.py (hunk 4 above)
Comment (Yshah112) [reply DashClothes/urls.py]: -change allauth  url to blank url 
-changed application url name to accounts

File: accounts/views.py (hunk 8 above)
Comment (Yshah112) [reply accounts/views.py]: changed to tempate classbase view

File: DashClothes/settings.py (hunk 7 above)
Comment (Yshah112) [reply DashClothes/settings.py:163]: added in .env file

File: accounts/models.py (hunk 9 above)
Comment (Yshah112) [reply accounts/models.py:15]: added docstring in feature/coomon

**Review by palakparmartntra — APPROVED** [review]

**Review by palakparmartntra — APPROVED** [review]

**Pull request merged**